### PR TITLE
Adds in Broken gunmods

### PR DESCRIPTION
**Pull request by zombiemimic**
## About The Pull Request
Adds in broken gun mods, rendering the affected guns inoperable.

## Why It's Good For The Game
joonk guns bad

## Changelog
:cl:
add: Added broken gunmods
/:cl:

### CODE REVIEW

**Comment by TheShown911** [comment]
I'm pretty sure this will affect guns printed in the autolathe, watch out.

**Comment by Alliostra** [comment]
This would be a noob trap, I'd recommend somehow making it glaringly obvious that you can't use a gun that has a broken mod, like add a "non-functional" in front of the gun name or something.

**Comment by MLGTASTICa** [comment]
Bad addition , doesn't fix any issues , makes the game overall less fun , forces even more people to break into chemistry. 
What is the point of the change ? what is it meant to adress as a game issue ? what is it meant to add ?

File: code/modules/projectiles/guns/mods/mods.dm [hunk 1]
```diff
@@ -406,7 +406,39 @@
 	I.gun_loc_tag = GUN_SCOPE
 	I.removable = FALSE
 
+//broken gun mods, for firearms beyond repair
+
+/obj/item/gun_upgrade/barrel/broken
+	name = "Plugged barrel"
```
Comment (Firefox13) [inline code/modules/projectiles/guns/mods/mods.dm:412]:
```suggestion
	name = "plugged barrel"
```

File: code/modules/projectiles/guns/mods/mods.dm [hunk 2]
```diff
@@ -406,7 +406,39 @@
...
+/obj/item/gun_upgrade/barrel/broken/New()
+	..()
+	var/datum/component/item_upgrade/I = AddComponent(/datum/component/item_upgrade)
+	I.weapon_upgrades = list(GUN_UPGRADE_BROKEN = TRUE)
+	I.destroy_on_removal = TRUE
+	I.gun_loc_tag = GUN_BARREL
+	I.removable = FALSE
+
+/obj/item/gun_upgrade/mechanism/broken
+	name = "Broken fire control group"
```
Comment (Firefox13) [inline code/modules/projectiles/guns/mods/mods.dm:427]:
```suggestion
	name = "broken fire control group"
```
what is even fire control group? receiver maybe?

Comment (zombiemimic) [reply code/modules/projectiles/guns/mods/mods.dm]: is now broken bolt

**Comment by Sparrowbird611** [comment]
I agree with MLG, this is a terrible addition. If you want less guns in maint, tweak the spawn rates, don't add this garbage.

**Comment by ValoTheValo** [comment]
This doesn't add anything of value to the game, what does Eris gain by having autolathe and maint guns randomly be inoperable?

**Pull request closed without merging**